radar/shared/src/types.rs

```diff
@@ -30,13 +30,7 @@ pub enum C4State {
     Carried,
 
     /// Bomb is currently actively ticking
-    Active {
-        /// Time remaining (in seconds) until detonation
-        time_detonation: f32,
-
-        /// Current bomb defuser
-        defuse: Option<BombDefuser>,
-    },
+    Active,
```

Comment:
I would keep the previous structure and move the bomb_site property, that allows us to remove the Option.

Like this:

    /// Bomb is dropped
    Dropped,

    /// Bomb is carried
    Carried,

    /// Bomb is currently actively ticking
    Active {
        /// Time remaining (in seconds) until detonation
        time_detonation: f32,

        /// Current bomb defuser
        defuse: Option<BombDefuser>,

        /// 0 = A
        /// 1 = B
        pub bomb_site: u8,
    },
    Active,

    /// Bomb has detonated
    Detonated {
        /// 0 = A
        /// 1 = B
        pub bomb_site: u8,
    },

    /// Bomb has been defused
    Defused {
        /// 0 = A
        /// 1 = B
        pub bomb_site: u8,
    },

Reply:
I was thinking of it, but then I had problem to access this in tsx.

Struct which you pruposing will lead to somethink like this

```json
{
   "state":{
      "active":{
         "time_detonation":23.19043,
         "defuse":{
            "timeRemaining":2.7998047,
            "playerName":"Waldo"
         }
      }
   },
   "bombSite":1
}
```
but for other statuses without additional struct it will be:
```json
{
   "state":"carried",
   "bombSite":null
}
```
For the first one, `{bomb.state}` will output error.
For the second one, `{bomb.state}` will output string.

As you can see, for your proposal, I can not think of a way how to access it in tsx. In my proposed struct, there is a unified way to accessing it, regardless of state. That way, there is no problem with error.

Maybe you know how to access such struct in tsx, then I will be glad to rework this.

Reply:
Yeah that's an issue I didn't notice.
But I think it has a solution.

I think we should be able to use: `#[serde(tag = "type", content = "data")]`
on the C4State to make the JSON format usable.

I can't test it now.

Reply:
I think the best would be an internally tagged enum representation in this case: https://serde.rs/enum-representations.html#internally-tagged

Reply:
As @WolverinDEV mentioned, the internally tagged enum is better in this case. I didn't read the documentation enough :)